### PR TITLE
Expand water virtual site tests

### PR DESCRIPTION
**Pull request by IAlibay**
Adds more checks #678 

TODO:
 - [x] Add test to check if the vs particle parameters are as expected.

## Developers certificate of origin
- [x] I certify that this contribution is covered by the MIT License [here](https://github.com/OpenFreeEnergy/openfe/blob/main/LICENSE) and the **Developer Certificate of Origin** at <https://developercertificate.org/>.

### CODE REVIEW

**Comment by IAlibay** [comment]
@hannahbaumann this is another good one to review if you want to get familiar with a specific section of the HTF.

File: openfe/tests/protocols/test_openmm_equil_rfe_protocols.py [hunk 1]
```diff
@@ -458,7 +461,58 @@ def test_dry_run_ligand_tip4p(benzene_system, toluene_system,
...
+        for entry in virtual_sites:
+            vs = htf.hybrid_system.getVirtualSite(entry)
+            vs_mass = htf.hybrid_system.getParticleMass(entry)
+            assert ensure_quantity(vs_mass, 'openff').m == pytest.approx(0)
+            vs_weights = [vs.getWeight(ix) for ix in range(vs.getNumParticles())]
+            np.testing.assert_allclose(
+                vs_weights, [0.786646558, 0.106676721, 0.106676721]
+            )
+            c, s, e = nonbond.getParticleParameters(entry)
+            assert ensure_quantity(c, 'openff').m == pytest.approx(-1.04844)
```
Comment (richardjgowers) [inline openfe/tests/protocols/test_openmm_equil_rfe_protocols.py:507]: should we look at contributing some sort of `assert_quantity_equal(val, expected_mag, expected_unit)` to openff units?

File: openfe/tests/protocols/test_openmm_equil_rfe_protocols.py [hunk 2]
```diff
@@ -458,7 +461,58 @@ def test_dry_run_ligand_tip4p(benzene_system, toluene_system,
     with tmpdir.as_cwd():
         sampler = dag_unit.run(dry=True)['debug']['sampler']
         assert isinstance(sampler, MultiStateSampler)
-        assert sampler._factory.hybrid_system
+
+        # Get the htf and check
```
Comment (richardjgowers) [inline openfe/tests/protocols/test_openmm_equil_rfe_protocols.py:465]: is it going to be a pain to make this a fixture and make these three tests individual?

Comment (IAlibay) [reply openfe/tests/protocols/test_openmm_equil_rfe_protocols.py:465]: Would have to try, but iirc tmpdir is function scoped, so you can't use it in a module scoped fixture - at least I remember this being a thing in the past. (definitely wouldn't recommend this being function scoped, it's super slow)

File: openfe/tests/protocols/test_openmm_equil_rfe_protocols.py (hunk 1 above)
Comment (IAlibay) [reply openfe/tests/protocols/test_openmm_equil_rfe_protocols.py:507]: If Matt wants to? We could also just make our own tooling if it's necessary.

File: openfe/tests/protocols/test_openmm_equil_rfe_protocols.py (hunk 2 above)
Comment (richardjgowers) [reply openfe/tests/protocols/test_openmm_equil_rfe_protocols.py:465]: ah yeah, there's [tmp_path_factory](https://docs.pytest.org/en/7.1.x/how-to/tmp_path.html#the-tmp-path-factory-fixture) which has longer scope

Comment (IAlibay) [reply openfe/tests/protocols/test_openmm_equil_rfe_protocols.py]: Done

**Review by richardjgowers — APPROVED** [review]

**Pull request merged**